### PR TITLE
Passthru Proxy Route/Config

### PR DESCRIPTION
**Pull request by dajinchu**
this makes it so the passthru proxy is attached to a config file so that you can mess with the routes, though it should probably always remain "/*"

Mostly this just gets rid of Vike logging an error in dev mode every time the passthru proxy is hit (it complains that no route matched)

### CODE REVIEW

File: bifrost-fastify/index.ts [hunk 1]
```diff
@@ -170,7 +171,7 @@ export const viteProxyPlugin: FastifyPluginAsync<
         // fwd.push(request.ip); TODO: not sure if this is needed
         headers["X-Forwarded-For"] = fwd.join(", ");
         headers["X-Forwarded-Host"] = host.host;
-        headers["X-Forwarded-Protocol"] = host.protocol;
+        headers["X-Forwarded-Proto"] = host.protocol;
```
Comment (dajinchu) [inline bifrost-fastify/index.ts:174]: slipped in this in- helps with https in rails

File: bifrost/proxy/pages/passthru/onRenderClient.tsx [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export default async function onRenderClient() {
+  // Passthru pages are unhandled by bifrost, reload to grab legacy content from server
+  window.location.reload();
```
Comment (dajinchu) [inline bifrost/proxy/pages/passthru/onRenderClient.tsx:3]: this behavior always happened but used to be implicit. vike would see navigation to a page it couldnt handle and reload

File: bifrost-fastify/index.ts [hunk 3]
```diff
@@ -199,9 +199,10 @@ export const viteProxyPlugin: FastifyPluginAsync<
           const location = reply.getHeader("location") as string;
           if (location) {
             const url = new URL(location);
-            if (url.host === upstream.host) {
+            if (url.host === upstream.host || url.host === host.host) {
               // rewrite redirect on upstream's host to the proxy host
               url.host = host.host;
+              url.protocol = host.protocol;
```
Comment (dajinchu) [inline bifrost-fastify/index.ts:205]: this also helps with rewriting rails redirects to https

File: tests/e2e/specs/http.spec.ts [hunk 4]
```diff
@@ -31,7 +31,8 @@ test.describe("requests", () => {
 
     test("returns wrapped proxy route when hit", async ({ request }) => {
       const req = await request.get("./custom");
-      expect(req.headers()["x-test-pageid"]).toBe("/proxy/pages");
+      const req = await request.get("./json-route");
+      expect(req.headers()["x-test-pageid"]).toBe("/proxy/pages/wrapped");
```
Comment (jacobto161) [inline tests/e2e/specs/http.spec.ts:35]: Nit: Do we expect this to be `/proxy/pages/passthru` in another case? If so, can we test that as well?

**Pull request merged**